sycl/test-e2e/Config/select_device.cpp

```diff
@@ -456,510 +386,274 @@ int main() {
-                 (sycl_be.find("level_zero") != std::string::npos)) ||
-                ((plt.get_backend() == backend::ext_oneapi_cuda) &&
-                 (sycl_be.find("cuda") != std::string::npos)) ||
-                ((plt.get_backend() == backend::ext_oneapi_hip) &&
-                 (sycl_be.find("hip") != std::string::npos))) {
-              fs << "DeviceName:{{" << name << "}},DriverVersion:HAHA{{" << ver
-                 << "}}" << std::endl;
-              passed = true;
-              break;
-            }
+          if (!dev.has(aspect::gpu))
+            continue;
+          std::string name = dev.get_info<info::device::name>();
+          addEscapeSymbolToSpecialCharacters(name);
+          std::string ver = dev.get_info<info::device::driver_version>();
+          if (is_known_be(plt.get_backend())) {
+            std::cout << "DeviceName:{{" << name << "}},DriverVersion:HAHA{{"
+                      << ver << "}}" << std::endl;
+            return 0;
           }
         }
       }
-      fs.close();
-    }
-  } else if (getenv("READ_DRIVER_MALFORMED_INFO")) {
-    std::ifstream fs;
-    fs.open("select_device_config.txt");
-    if (fs.is_open()) {
-      std::string allowlist;
-      std::getline(fs, allowlist);
-      if (!allowlist.empty()) {
-        setenv("SYCL_DEVICE_ALLOWLIST", allowlist.c_str(), 0);
-        std::vector<DevDescT> components(getAllowListDesc(allowlist));
-        std::cout << "SYCL_DEVICE_ALLOWLIST=" << allowlist << std::endl;
-
-        try {
-          sycl::queue deviceQueue(gpu_selector_v);
-          device dev = deviceQueue.get_device();
-          const auto &plt = dev.get_platform();
-        } catch (sycl::exception &E) {
-          // Workaround to make CI pass.
-          // TODO: after the submission of PR intel/llvm:3826, create PR to
-          // intel/llvm-test-suite with removal of 1st parameter of the vector,
-          // and transformation of std::vector<std::string> to std::string
-          const std::vector<std::string> expectedMsgs{
-              "Malformed syntax in SYCL_DEVICE_ALLOWLIST",
-              "Key DriverVersion of SYCL_DEVICE_ALLOWLIST should have value "
-              "which starts with {{"};
-          const std::string gotMessage(E.what());
-          for (const auto &expectedMsg : expectedMsgs) {
-            if (gotMessage.find(expectedMsg) != std::string::npos) {
-              passed = true;
-              break;
-            }
-            passed = false;
-          }
-        }
+      return 1;
+    } else {
+      try {
+        sycl::queue deviceQueue(gpu_selector_v);
+        device dev = deviceQueue.get_device();
+        const auto &plt = dev.get_platform();
+      } catch (sycl::exception &E) {
+        std::cout << "Caught exception: " << E.what() << std::endl;
+        if (E.what() ==
+            "Key DriverVersion of SYCL_DEVICE_ALLOWLIST should have value which starts with {{ -30 (PI_ERROR_INVALID_VALUE)"sv)
+          return 0;
```

Comment:
This (and similar changes in few other places) addresses TODO/workaround from the LHS.